typescript/infra/scripts/check/check-warp-deploy.ts

```diff
@@ -36,23 +35,39 @@ async function main() {
   ];
 
   const registries = [DEFAULT_REGISTRY_URI];
+  const registry = getRegistry({
+    registryUris: registries,
+    enableProxy: true,
+  });
+
   const warpCoreConfigMap =
     await getWarpConfigMapFromMergedRegistry(registries);
 
   console.log(chalk.yellow('Skipping the following warp routes:'));
   routesToSkip.forEach((route) => console.log(chalk.yellow(`- ${route}`)));
 
-  const warpIdsToCheck = Object.keys(warpCoreConfigMap).filter(
-    (warpRouteId) => !routesToSkip.includes(warpRouteId),
-  );
-
-  // Get all the chains from warpCoreConfigMap. Used to initialize the MultiProvider.
-  const warpConfigChains = warpIdsToCheck.reduce((chains, warpRouteId) => {
-    const warpConfigs = warpCoreConfigMap[warpRouteId];
-    assert(warpConfigs, `Config not found in registry for ${warpRouteId}`);
-    Object.keys(warpConfigs).forEach((chain) => chains.add(chain));
-    return chains;
-  }, new Set<ChainName>());
+  const isTestnetRoute = async (warpRouteConfig: any) => {
+    for (const chain of Object.keys(warpRouteConfig)) {
+      const chainMetadata = await registry.getChainMetadata(chain);
+      if (chainMetadata?.isTestnet) return true;
+    }
+    return false;
+  };
+
+  const warpConfigChains = new Set<ChainName>();
+  const warpIdsToCheck = await Promise.all(
+    Object.keys(warpCoreConfigMap).map(async (warpRouteId) => {
+      const warpRouteConfig = warpCoreConfigMap[warpRouteId];
+      const isTestnet = await isTestnetRoute(warpRouteConfig);
+      if (!routesToSkip.includes(warpRouteId) && !isTestnet) {
+        Object.keys(warpRouteConfig).forEach((chain) =>
+          warpConfigChains.add(chain),
+        );
+        return warpRouteId;
+      }
+      return null;
+    }),
+  ).then((results) => results.filter((id) => id !== null));
```

Comment:
maybe this should instead look at the `environment` - and ensure we filter out testnets on mainnet3 but _only_ do testnets on testnet4